File: old/gen3.c
#include <stdio.h>
#include <stdint.h>

int main() {
    FILE *file = fopen("sequence.ft", "wb");
    if (!file) {
        perror("Failed to open file");
        return 1;
    }

    // Definisci la sequenza di bit da trasmettere
    char *bits = "100010001000111010001110100011101000111010001110100010001000111010001110100011101000111011101110100000000000000000000000000000000000000000000100010001000111010001110100011101000111010001110100010001000111010001110100011101000111010001000100000000000000000000000000000000000";
    double frequency = 434030000; // Frequenza per i bit '1'
    uint32_t duration = 350000;   // 350 microsecondi in nanosecondi
    uint32_t padding = 0;

    for (int i = 0; bits[i] != '\0'; i++) {
        double freq_to_write = bits[i] == '1' ? frequency : 0;  // Usa 0 Hz per '0'
        
        // Scrivi la frequenza
        fwrite(&freq_to_write, sizeof(double), 1, file);
        // Scrivi la durata
        fwrite(&duration, sizeof(uint32_t), 1, file);
        // Scrivi il padding
        fwrite(&padding, sizeof(uint32_t), 1, file);
    }

    fclose(file);
    printf("File written successfully.\n");
    return 0;
}

File: old/gen3.py
import struct

# Parametri
frequency_hz = 433944000  # Frequenza di trasmissione in Hz per bit '1'
duration_ns = 350000  # Durata di ogni bit in nanosecondi
sequence = "1000100010001110100011101000111010001110100011101000100010001110100011101000111010001110111011101"
sequence2 ="1000100010001110100011101000111010001110100011101000100010001110100011101000111010001110100010001"

freq_bytes = struct.pack('d', frequency_hz)  # 'd' per double precision float

# Creazione del file .ft
with open('doorbell.ft', 'wb') as f:
    # Il primo chunk potrebbe essere un valore di offset, se necessario
    # Per semplicità, assumiamo che non ci sia un offset iniziale necessario
    for bit in sequence:
        if bit == '1':
            amplitude = struct.pack('f', 10)  # 'd' per double precision float
        else:
            amplitude = struct.pack('f', 0)  # 'd' per double precision float
        
        duration_bytes = struct.pack('I', duration_ns)  # 'I' per unsigned int
        padding_bytes = struct.pack('I', 0)  # 4 byte di padding
        
        f.write(freq_bytes + duration_bytes + amplitude + padding_bytes)  # Scrivi il chunk completo

    for i in range(1,25000):
        amplitude = struct.pack('f', 0)  # 'd' per double precision float
        duration_bytes = struct.pack('I', duration_ns)  # 'I' per unsigned int
        padding_bytes = struct.pack('I', 0)  # 4 byte di padding
        f.write(freq_bytes + duration_bytes + amplitude + padding_bytes)  # Scrivi il chunk completo

    for bit in sequence2:
        if bit == '1':
            amplitude = struct.pack('f', 10)  # 'd' per double precision float
        else:
            amplitude = struct.pack('f', 0)  # 'd' per double precision float

        duration_bytes = struct.pack('I', duration_ns)  # 'I' per unsigned int
        padding_bytes = struct.pack('I', 0)  # 4 byte di padding

        f.write(freq_bytes + duration_bytes + amplitude + padding_bytes)  # Scrivi il chunk completo

    for i in range(1,25000): 
        amplitude = struct.pack('f', 0)  # 'd' per double precision float
        duration_bytes = struct.pack('I', duration_ns)  # 'I' per unsigned int
        padding_bytes = struct.pack('I', 0)  # 4 byte di padding
        f.write(freq_bytes + duration_bytes + amplitude + padding_bytes)  # Scrivi il chunk completo


print("File .ft creato con successo.")
